astropy/modeling/physical_models.py

```diff
@@ -0,0 +1,187 @@
+            Blackbody spectrum. The units are determined from the units of
+            ``scale``.
+
+        .. note::
+
+            Use `numpy.errstate` to suppress Numpy warnings, if desired.
+
+        .. warning::
+
+            Output values might contain ``nan`` and ``inf``.
+
+        Raises
+        ------
+        ValueError
+            Invalid temperature.
+
+        ZeroDivisionError
+            Wavelength is zero (when converting to frequency).
+        """
+        if not isinstance(temperature, u.Quantity):
+            in_temp = u.Quantity(temperature, u.K)
+        else:
+            in_temp = temperature
+
+        # Convert to units for calculations, also force double precision
+        with u.add_enabled_equivalencies(u.spectral() + u.temperature()):
+            freq = u.Quantity(x, u.Hz, dtype=np.float64)
+            temp = u.Quantity(in_temp, u.K)
+
+        # check the units of scale and setup the output units
+        bb_unit = u.erg / (u.cm ** 2 * u.s * u.Hz * u.sr)  # default unit
+        if hasattr(scale, "unit"):
+            # check that the units on scale are covertable to surface brightness units
+            if not scale.unit.is_equivalent(bb_unit, u.spectral_density(x)):
+                raise ValueError(f"scale units not surface brightness: {scale.unit}")
+            mult_scale = scale.value
+            bb_unit = scale.unit
+            print(bb_unit)
+        else:
+            mult_scale = scale
+
+        # Check if input values are physically possible
+        if np.any(temp < 0):
+            raise ValueError(f"Temperature should be positive: {temp}")
+        if not np.all(np.isfinite(freq)) or np.any(freq <= 0):
+            warnings.warn(
+                "Input contains invalid wavelength/frequency value(s)",
+                AstropyUserWarning,
+            )
+
+        log_boltz = const.h * freq / (const.k_B * temp)
+        boltzm1 = np.expm1(log_boltz)
+
+        # Calculate blackbody flux
+        bb_nu = 2.0 * const.h * freq ** 3 / (const.c ** 2 * boltzm1) / u.sr
+
+        y = mult_scale * bb_nu.to(bb_unit, u.spectral_density(freq))
+
+        # If the temperature parameter has no unit, we should return a unitless
+        # value. This occurs for instance during fitting, since we drop the
+        # units temporarily.
+        if hasattr(temperature, "unit"):
+            return y
+        else:
+            return y.value
+
+    @property
+    def input_units(self):
+        # The input units are those of the 'x' value, which should always be
+        # Hz. Because we do this, and because input_units_allow_dimensionless
+        # is set to True, dimensionless values are assumed to be in Hz.
+        return {"x": u.Hz}
+
+    def _parameter_units_for_data_units(self, inputs_unit, outputs_unit):
+        return OrderedDict([("temperature", u.K)])
+
+    @property
+    def bolometric_flux(self):
+        """Bolometric flux."""
+        return const.sigma_sb * self.temperature ** 4 / np.pi
```

Comment:
Hmm, so the bolometric flux does not take the fitted scale into account? Talking to you the other week, I was under the impression that this would be derived from both temperature and scale.

Reply:
Good catch!  Scale definitely should be used.  Added it.